lib/widgets/action_sheet.dart

```diff
@@ -181,6 +142,264 @@ class MessageActionSheetCancelButton extends StatelessWidget {
   }
 }
 
+/// Show a sheet of actions you can take on a topic.
+void showTopicActionSheet(BuildContext context, {
+  required int channelId,
+  required String topic,
+}) {
+  final narrow = TopicNarrow(channelId, topic);
+  UserTopicUpdateButton button({
+    UserTopicVisibilityPolicy? from,
+    required UserTopicVisibilityPolicy to,
+  }) {
+    return UserTopicUpdateButton(
+      currentVisibilityPolicy: from,
+      newVisibilityPolicy: to,
+      narrow: narrow,
+      pageContext: context);
+  }
+
+  final mute =                 button(to:   UserTopicVisibilityPolicy.muted);
+  final unmute =               button(from: UserTopicVisibilityPolicy.muted,
+                                      to:   UserTopicVisibilityPolicy.none);
+  final unmuteInMutedChannel = button(to:   UserTopicVisibilityPolicy.unmuted);
+  final follow =               button(to:   UserTopicVisibilityPolicy.followed);
+  final unfollow =             button(from: UserTopicVisibilityPolicy.followed,
+                                      to:   UserTopicVisibilityPolicy.none);
```

Comment:
Web's radio-buttons approach seems like a fine solution to me, but again we might postpone that; we should make an issue if it's planned though. @gnprice, what do you think?

Reply:
Yeah, this PR follows the state transitions in the mobile app very closely, including the UX. If we are going with a newer radio-button design, it will probably save some time to get this right with a Figma redesign.

See [discussion here](https://chat.zulip.org/#narrow/channel/48-mobile/topic/Adding.20user.20topic.20visibility.20markers.20.28muted.2C.20default.2C.20etc.2E.29/near/1969293).

Reply:
Oh I see, that discussion is helpful context! I missed it in my review because I didn't see it linked from this PR or the issue. 🙂 I agree with the outcome there:

> Yeah, some version of that would probably be good to put in mobile's action sheet too.
> 
> Definitely post-launch, though.

Would you make an issue for that? It can link to this discussion, to memoize the work of figuring out what state transitions we're missing. (I used a lot of post-it notes, haha.)

Reply:
Sure! Opened #1078.

Reply:
Thanks!